applications/ContactStructuralMechanicsApplication/custom_processes/master_slave_process.cpp

```diff
@@ -53,22 +53,22 @@ void MasterSlaveProcess::Execute()
         
         #pragma omp for
         for(int i = 0; i < num_conditions; ++i) {
-            auto it_cond = conditions_array.begin() + i;
-            const auto& this_geometry = it_cond->GetGeometry();
+            auto it_cond = it_cond_begin + i;
+            const auto& r_geometry = it_cond->GetGeometry();
             
             // We set the condition as master or slave (master by default)
             bool is_interface = true;
             bool is_slave = true;
-            for (IndexType i_node = 0; i_node < this_geometry.size(); ++i_node) {
-                auto& this_node = this_geometry[i_node];
-                if (this_node.IsDefined(INTERFACE)) {
-                    if (!this_node.Is(INTERFACE)) is_interface = false;
-                    else if (!this_node.Is(SLAVE)) is_slave = false;
+            for (IndexType i_node = 0; i_node < r_geometry.size(); ++i_node) {
+                const auto& r_node = r_geometry[i_node];
+                if (r_node.IsDefined(INTERFACE)) {
+                    if (r_node.IsNot(INTERFACE)) is_interface = false;
+                    else if (r_node.IsNot(SLAVE)) is_slave = false;
```

Comment:
is this threadsafe? I think not

Reply:
I will check, in any case is not used always

Reply:
thats the point
if the conditions share a node, then the node might be accessed twice from two threads

Reply:
Yeah, but accessing is threadsafe, writing is not, right @roigcarlo ?

Reply:
ah sorry you are only accessing, confused myself.
Nevermind, sorry for the noise